### ISSUE TITLE
Ensure 16-bit atomics are used properly on aarch64

### ISSUE THREAD

**Issue by mbautin** [yugabyte/yugabyte-db#9196]

**Comment by bmatican** [comment]
Closing in favor of #11412 where Sergei validated that this will work fine once we upgrade boost

**Comment by spolitov** [comment]
boost was not yet upgraded in yugabyte-db repo

**Comment by bmatican** [comment]
@spolitov no worries, then please close both once you land that
